common4j/src/main/com/microsoft/identity/common/java/logging/Logger.java

```diff
@@ -284,6 +289,24 @@ public static void info(final String tag,
         log(tag, Logger.LogLevel.INFO, null, message, null, false);
     }
 
+    /**
+     * Send a {@link Logger.LogLevel#INFO} log message without PII.
+     *
+     * @param tag     Used to identify the source of a log message. It usually identifies the class
+     *                or activity where the log call occurs.
+     * @param message The message to log.
+     * @param object The object to be printed.
+     */
+    public static void infoWithObject(final String tag,
+                            final String message,
+                            final Logging object) {
+        if (isAllowPii()) {
+            log(tag, Logger.LogLevel.INFO, null, message, object.toSafeString(true), null, object.containsPii());
+        } else {
+            log(tag, Logger.LogLevel.INFO, null, message, object.toSafeString(false), null, false);
```

Comment:
Why don't `log(tag, Logger.LogLevel.INFO, null, message, object.toSafeString(false), null, object.containsPii());`

Reply:
Good question. In this case, the developer has set that they don't want PII to be logged. `object.containsPii()` returns `true` if it contains fields that are PII. So we don't want to use this method, as that would make the value `containsPii` true, which is a contradiction of the configuration the developer set. 

Maybe I should rethink this approach, but my goal was to be able to distinguish between 3 cases: 
1. developer doesn't want PII to be logged. No PII should be logged (`object.toSafeString(mayContainPii = false)` and `containsPii = false`).
2. developer wants PII to be logged (`object.toSafeString(mayContainPii = false)`)
   2a. The log actually contains PII - `object.containsPii()` returns `true`, which sets `containsPii` to `true`.
   2b. The log doesn't contain PII - `object.containsPii()` returns `false`, which sets `containsPii` to `false`.
   
 Developers may want to do different things depending on the value of `containsPii`.

Reply:
I guess `object.containsPii()` indicates whether the object contains PII information.  `mayContainPii` determines which version of logging may need to be used and always remains the same as` isAllowPii()`. 

1. Developer doesn't want PII to be logged (`isAllowPii()=false`) 
1a. Object has PII info(`containsPii=true`).  -> mayContainPii=false, choose the version without PII -> void log return. There is no logging.     **That's not the ideal outcome. Thus use false instead of object.containsPii()**
1b. Object doesn't have PII info(`containsPii=false`). ->  no matter the value of mayContainPii, the log should be the same.  -> Log without PII info.

2. Developer wants PII to be logged (`isAllowPii()=true`)
2a.  Object has PII info(`containsPii=true`). -> mayContainPii=true, choose the PII version -> Log with PII info.
2b. Object doesn't have PII info(`containsPii=false`) ->  no matter the value of mayContainPii, the log should be the same. -> Log without PII info.

I understand why the `'false'` is set up here.

Reply:
Exactly. In the case of 1a, we can't print any PII if `isAllowPii` is set to `false`. Thus, we use the `toSafeString(mayContainPii=false)`, which should return a string that contains no PII. Thus, by default, `containsPii` will always be `false`.